edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/RouteConfigurationConverter.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Core.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Org.BouncyCastle.Security;
+
+    public class RouteConfigurationConverter : JsonConverter<IDictionary<string, RouteConfiguration>>
+    {
+        public override void WriteJson(JsonWriter writer, IDictionary<string, RouteConfiguration> value, JsonSerializer serializer) => throw new NotSupportedException();
+
+        public override IDictionary<string, RouteConfiguration> ReadJson(JsonReader reader, Type type, IDictionary<string, RouteConfiguration> value, bool hasValue, JsonSerializer serializer)
+        {
+            var routes = new Dictionary<string, RouteConfiguration>();
+            JToken token = JToken.Load(reader);
+
+            foreach (JToken child in token.Children())
+            {
+                // The route name must not be empty
+                Preconditions.CheckNonWhiteSpace(child.First.Path, nameof(child.First.Path));

```

Comment:
Suggest to provide more detailed message, e.g. print out which children node.

Reply:
CheckNonWhiteSpace doesn't take a custom string message.

---
In reply to: [377443190](https://github.com/Azure/iotedge/pull/2469#discussion_r377443190) [](ancestors = 377443190)